qemu/tests/rng_stress.py

```diff
@@ -0,0 +1,87 @@
+import logging
+import aexpect
+import time
+
+from autotest.client import utils
+
+from virttest import error_context
+from virttest import utils_test
+from virttest.qemu_devices import qdevices
+from avocado.core import exceptions
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    1) Boot up vm with one or more rng devices
+    2) Read from /dev/random in host (optional)
+    3) Run random read in guest
+    4) Switch rng device if there is more than on rng devices
+    5) Run random read in guest
+    6) Read from /dev/random in host (optional)
+    7) Clean random read in host (optional)
+    """
+
+    def get_rng_list(vm):
+        """
+        Get attached rng devices from device dictionary
+        """
+        rng_list = []
+        for device in vm.devices:
+            if isinstance(device, qdevices.QDevice):
+                if device.get_param("driver") == "virtio-rng-pci":
+                    rng_list.append(device)
+        return rng_list
+
+    def get_available_rng(session):
+        """
+        Get available rng devices from /sys/devices
+        """
+        verify_cmd = params["driver_available_cmd"]
+        try:
+            rng_devices = session.cmd_output(verify_cmd)
```

Comment:
After test, found the command is also contained in the output. 
06:29:40 DEBUG| Sending command: cat /sys/devices/virtual/misc/hw_random/rng_available
06:29:40 DEBUG| ###rng_devices=cat /sys/devices/virtual/misc/hw_random/rng_availab<at /sys/devices/virtual/misc/hw_random/rng_availabl                         ^H^H^H^H^H^H^H^H^H^H^H^H^H^H^H^H^H^H^H^H^H^H^H^H^He
06:29:40 DEBUG| virtio_rng.0
06:29:40 DEBUG|
06:29:40 DEBUG| ###rng_devices=['cat', '/sys/devices/virtual/misc/hw_random/rng_availab<at', '/sys/devices/virtual/misc/hw_random/rng_availabl', '\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08\x08e', 'virtio_rng.0']
06:29:40 DEBUG| ###rng_list=[<virttest.qemu_devices.qdevices.QDevice object at 0x5a7fe10>]